src/web/mjs/connectors/templates/MadTheme.mjs

```diff
@@ -0,0 +1,74 @@
+import Connector from '../../engine/Connector.mjs';
+import Manga from '../../engine/Manga.mjs';
+export default class MadTheme extends Connector {
+    constructor() {
+        super();
+        super.id = undefined;
+        super.label = undefined;
+        this.tags = [];
+        this.url = undefined;
+        this.path = '/az-list';
+        this.queryMangaTitleFromURI = 'div.name.box h1';
+        this.queryMangas = 'div.book-detailed-item div.thumb a';
+        this.queryPages = 'div.chapter-image';
+        this.queryChapterTitle = 'strong.chapter-title';
+    }
+    async _getMangaFromURI(uri) {
+        const request = new Request(new URL(uri), this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangaTitleFromURI);
+        const id = uri.pathname;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+    async _getMangas() {
+        const mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangas.length ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.path + '?page=' + page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: element.pathname,
+                title: element.title.trim()
+            };
+        });
+    }
+    async _getChapters(manga) {
+        let mangaid = manga.id.split('/').pop();
+        mangaid = '/'+mangaid;//make sure to take last part of url for the api
+        const uri = new URL('/api/manga'+mangaid+'/chapters?source=detail', this.url);

```

Comment:
`mangaid = '/'+mangaid;` extra instruction adds no value : use `/api/manga/` in the URL